File: backend/src/main/java/com/realestate/agent/service/RoleService.java
package com.realestate.agent.service;

import com.realestate.agent.dto.RoleRequest;
import com.realestate.agent.dto.RoleResponse;

import java.util.List;

public interface RoleService {

    List<RoleResponse> getAllRoles();

    RoleResponse getRoleById(Long roleId);

    RoleResponse createRole(RoleRequest request);

    RoleResponse updateRole(Long roleId, RoleRequest request);

    void assignRoleToUser(Long roleId, Long userId);

    void removeRoleFromUser(Long roleId, Long userId);

    void deleteRole(Long roleId);
}

File: frontend/real-estate-frontend/src/services/roleService.js
import apiClient from "./apiClient";

// Get all system roles from PostgreSQL
export const getAllRoles = async () => {
  return apiClient.get("/api/admin/roles");
};

// Get role by ID
export const getRoleById = async (roleId) => {
  return apiClient.get(`/api/admin/roles/${roleId}`);
};

// Create a new role
export const createRole = async (roleData) => {
  return apiClient.post("/api/admin/roles", roleData);
};

// Update role details
export const updateRole = async (roleId, roleData) => {
  return apiClient.put(`/api/admin/roles/${roleId}`, roleData);
};

// Assign role to user
export const assignRoleToUser = async (roleId, userId) => {
  return apiClient.post(`/api/admin/roles/${roleId}/assign/${userId}`);
};

// Remove role from user
export const removeRoleFromUser = async (roleId, userId) => {
  return apiClient.delete(`/api/admin/roles/${roleId}/users/${userId}`);
};

// Delete role
export const deleteRole = async (roleId) => {
  return apiClient.delete(`/api/admin/roles/${roleId}`);
};
